pycheribuild/projects/run_qemu.py

```diff
@@ -585,6 +639,52 @@ def dependencies(cls, config: CheriConfig) -> "list[str]":
         return result
 
 
+class AbstractLaunchFreeBSDProgram(LaunchBsdUserQEMUBase):
+    do_not_add_to_targets = True
+    _source_class = BuildCHERIBSD
+    _freebsd_class = BuildCHERIBSD
+    _always_add_suffixed_targets = True
+    supported_architectures = [CompilationTargets.CHERIBSD_RISCV_PURECAP]
+
+    def __init__(self, config):
+        super().__init__(config)
+        self.rootfs_path = self._source_class.get_rootfs_dir(self, config=config)
+
+    @classmethod
+    def setup_config_options(cls, **kwargs):
+        super().setup_config_options(**kwargs)
+        cls.chroot = cls.add_bool_option("chroot", default=False, show_help=True,
+                                         help="Change the root directory to a sysroot before executing a command.")
+        cls.jail = cls.add_bool_option("jail", default=False, show_help=True,
+                                       help="Enter a jail with a sysroot before executing a command.")
+        cls.jail_extra_args = cls.add_config_option("jail-extra-args", metavar="ARGS", kind=list, show_help=True,
+                                                    help="Extra jail parameters to pass to jail(8).")
+
+    @classmethod
+    def dependencies(cls: "typing.Type[_RunMultiArchFreeBSDImage]", config: CheriConfig) -> "list[str]":
+        xtarget = cls.get_crosscompile_target(config)
+        result = ["bsd-user-qemu", cls._source_class.get_class_for_target(xtarget).target]
+        return result
+
+
+class LaunchCheriBSDShell(AbstractLaunchFreeBSDProgram):
+    target = "run-shell"
+
+    def __init__(self, config):
+        super().__init__(config)
+        self.command = None
```

Comment:
Probably a sign some of the behaviour should be pushed down into LaunchCheriBSDExec